relayer/message_relayer.go

```diff
@@ -71,10 +73,27 @@ func newMessageRelayer(
 		)
 		return nil, err
 	}
+	var signingSubnet ids.ID
+	if relayer.sourceSubnetID == constants.PrimaryNetworkID {
+		// If the message originates from the primary subnet, then we instead "self sign" the message using the validators of the destination subnet.
+		signingSubnet, err = relayer.pChainClient.ValidatedBy(context.Background(), destinationBlockchainID)
```

Comment:
Message relayers are coupled to the source blockchain config, not the destination. We get the destination blockchain from the Teleporter message itself, but then have to look up the corresponding destination subnet.

Reply:
Yeah, I'm just saying we could look up the destination subnet because all the pairings are set in the config to avoid the API call. If it seems like more trouble than it's worth that's okay

Reply:
Oh, gotcha. I misunderstood your original message. 

That makes a lot of sense, and reducing the number of RPC calls is an easy optimization. I'll include that in this PR.

Reply:
Added